### PR TITLE
Port to fn component

### PR DESCRIPTION
**Pull request by cquiroz**
Signed-off-by: Carlos Quiroz <carlos.quiroz@noirlab.edu>

### CODE REVIEW

File: explore/src/main/scala/explore/targeteditor/SiderealTargetEditor.scala [hunk 1]
```diff
@@ -74,198 +74,191 @@ object SiderealTargetEditor {
...
-          { t: SiderealTarget =>
-            EditSiderealInput.name.replace(t.name.assign) >>>
-              TargetQueries.UpdateSiderealTracking(t.tracking) >>>
-              TargetQueries.replaceMagnitudes(t.magnitudes)
-          }
-        )
+  val component =
+    ScalaFnComponent
+      .withHooks[Props]
+      .renderWithReuse { props =>
```
Comment (rpiaggio) [inline explore/src/main/scala/explore/targeteditor/SiderealTargetEditor.scala:80]: You should be able to do`ScalaFnComponent.withReuse[Props]` if there are no hooks.

**Review by rpiaggio — APPROVED** [review]
LGTM

File: explore/src/main/scala/explore/targeteditor/SiderealTargetEditor.scala (hunk 1 above)
Comment (cquiroz) [reply explore/src/main/scala/explore/targeteditor/SiderealTargetEditor.scala:80]: Oh didn't know that

**Review by toddburnside — APPROVED** [review]
LGTM

**Pull request merged**